org.hl7.fhir.publisher.core/src/main/java/org/hl7/fhir/igtools/publisher/AIProcessor.java

```diff
@@ -0,0 +1,429 @@
+    NpmPackage npm = NpmPackage.fromPackage(new FileInputStream(Utilities.path(rootDir, "package.tgz")));
+
+    for (String lang : langs) {
+      processFolderNew(new File(Utilities.path(rootDir, lang)), npm, lang);
+    }
+  }
+
+  public void processOldTemplates() throws IOException {
+    NpmPackage npm = NpmPackage.fromPackage(new FileInputStream(Utilities.path(rootDir, "package.tgz")));
+    processFolderOld(new File(rootDir), npm);
+  }
+
+  public void processFolderNew(File dir, NpmPackage npm, String lang) throws IOException {
+    List<LoadedResource> resources = new ArrayList<>();
+    List<Page> pages = new ArrayList<>();
+
+    // list all the resources in the IG, and note their type and whether they're an example
+    loadResources(dir, resources, npm, true);
+
+    // now list all the pages in the IG, and decide whether they are a page, a resource, or an alternate view of a resource
+    long oldSize = loadPages(dir, pages, resources);
+    long newSize = 0;
+
+    ZipGenerator zip = new ZipGenerator(Utilities.path(dir, "ai.zip"));
+    StringBuilder llms = new StringBuilder();
+    llms.append("# " + npm.vid() + ": " + npm.title() + "("+lang+")\r\n\r\n");
+    llms.append("## Pages\r\n\r\n");
+    for (Page p : pages) {
+      if (p.f.getName().equals("index.html")) {
+        newSize += produceMDForPage(llms, p, zip, npm);
+      }
+    }
+    for (Page p : pages) {
+      if (!p.f.getName().equals("index.html") && p.kind == PageKind.PAGE) {
+        newSize += produceMDForPage(llms, p, zip, npm);
+      }
+    }
+    llms.append("\r\n## Resources\r\n");
+    resources.sort(new LoadedResourceSorter());
+    String t = null;
+    for (LoadedResource r : resources) {
+      Page p = getPageForResource(r, pages);
+      if (!r.type.equals(t)) {
+        t = r.type;
+        llms.append("\r\n### " + t.substring(1) + "\r\n\r\n");
+      }
+      String bn = r.filename;
+      r.filename = r.filename.replace(".html", ".ai.html");
+      newSize += produceMDForResource(llms, r, p, dir, zip, true);
+      file(dir, r.filename).delete();
+      file(dir, r.filename, ".md").renameTo(file(dir, bn, ".md"));
+    }
+    zip.addFileSource("llms.txt", llms.toString(), false);
+    zip.addFileSource("llms.md", llms.toString(), false);
+    zip.close();
+    FileUtilities.stringToFile(llms.toString(), Utilities.path(dir, "llms.txt"));
+    newSize += llms.toString().length();
+    String pct = oldSize == 0 ? "n/a" : "" + (100 - ((newSize * 100) / oldSize));
+    System.out.println("AI format for " + dir.getAbsolutePath() + " - % reduction = " + pct);
+  }
+
+  private File file(File dir, String s, String ext) throws IOException {
+    return ManagedFileAccess.file(Utilities.path(dir, FileUtilities.changeFileExt(s, ext)));
+  }
+
+  private File file(File dir, String s) throws IOException {
+    return ManagedFileAccess.file(Utilities.path(dir, s));
+  }
+
+  public void processFolderOld(File dir, NpmPackage npm) throws IOException {
```

Comment:
Add some comments for what 'process' means, or change the name to be clearer.

The following would be a little too verbose, but at least a comment to indicate intent on these public methods would give some hints for future development.

generateMarkdownForPagesAndIndexInLLMS...